src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -580,78 +580,198 @@ std::vector<ParsedQuery> Visitor::visit(Parser::UpdateContext* ctx) {
+  parsedQuery_._clause = visitAlternative<parsedQuery::UpdateClause>(
+      ctx->load(), ctx->insertData(), ctx->deleteData());
+  parsedQuery_.datasetClauses_ = activeDatasetClauses_;
+  return {std::move(parsedQuery_)};
 }
 
 // ____________________________________________________________________________________
 Load Visitor::visit(Parser::LoadContext* ctx) {
-  return Load{
-      static_cast<bool>(ctx->SILENT()), visit(ctx->iri()),
-      ctx->graphRef() ? visit(ctx->graphRef()) : std::optional<GraphRef>{}};
+  reportNotSupported(ctx, "LOAD Update is");
 }
 
+// Helper functions for some inner parts of graph management operations.
+namespace {
 // ____________________________________________________________________________________
-Clear Visitor::visit(Parser::ClearContext* ctx) {
-  return Clear{static_cast<bool>(ctx->SILENT()), visit(ctx->graphRefAll())};
+// Transform a `GraphRefAll` or `GraphOrDefault` into a
+// `SparqlTripleSimpleWithGraph::Graph`/`parsedQuery::GroupGraphPattern::GraphSpec`
+// (which are the same type).
+SparqlTripleSimpleWithGraph::Graph transformGraph(const auto& graph) {
+  using Graph = SparqlTripleSimpleWithGraph::Graph;
+  return std::visit(
+      ad_utility::OverloadCallOperator{
+          [](const ad_utility::triple_component::Iri& iri) -> Graph {
+            return iri;
+          },
+          [](const ALL&) -> Graph { return Variable("?g"); },
+          [](const DEFAULT&) -> Graph { return std::monostate{}; },
+          [](const NAMED&) -> Graph { AD_FAIL(); }},
+      graph);
 }
 
 // ____________________________________________________________________________________
-Drop Visitor::visit(Parser::DropContext* ctx) {
-  return Drop{static_cast<bool>(ctx->SILENT()), visit(ctx->graphRefAll())};
+// Make a `GraphPatternOperation` that matches all triples in the graph.
+GraphPatternOperation makeAllTripleGraphPattern(const auto& graph) {
+  GraphPattern inner;
+  inner._graphPatterns.emplace_back(BasicGraphPattern{
+      {{{Variable("?s")}, Variable("?p"), {Variable("?o")}}}});
+  return {
+      parsedQuery::GroupGraphPattern{std::move(inner), transformGraph(graph)}};
 }
 
 // ____________________________________________________________________________________
-Create Visitor::visit(Parser::CreateContext* ctx) {
-  return Create{static_cast<bool>(ctx->SILENT()), visit(ctx->graphRef())};
+// Make a `SparqlTripleSimpleWithGraph` that templates all triples in the graph.
+SparqlTripleSimpleWithGraph makeAllTripleTemplate(const auto& graph) {
+  return {{Variable("?s")},
+          {Variable("?p")},
+          {Variable("?o")},
+          transformGraph(graph)};
 }
+}  // namespace
 
 // ____________________________________________________________________________________
-Add Visitor::visit(Parser::AddContext* ctx) {
+ParsedQuery Visitor::visit(Parser::ClearContext* ctx) {
+  return makeClear(visit(ctx->graphRefAll()));
+}
+
+ParsedQuery Visitor::makeClear(const GraphOrDefault& graph) {
```

Comment:
Comment that what you do in there is the internal representation of `DELETE {?s ?p ?o} WHERE {?s ?p ?o}`

(similar for the other transfoirmations below), then this is much easier to read.